src/django_components/component.py

```diff
@@ -2937,8 +3048,30 @@ def _render_impl(
         #       but instead can render one component at a time.
         #############################################################################
 
-        with _prepare_template(component, template_data) as template:
-            component_ctx.template_name = template.name
+        # TODO_v1 - Currently we have to pass `template_data` to `prepare_component_template()`,
+        #     so that `get_template_string()`, `get_template_name()`, and `get_template()`
+        #     have access to the data from `get_template_data()`.
+        #
+        #     Because of that there is one layer of `Context.update()` called inside `prepare_component_template()`.
+        #
+        #     Once `get_template_string()`, `get_template_name()`, and `get_template()` are removed,
+        #     we can remove that layer of `Context.update()`, and NOT pass `template_data`
+        #     to `prepare_component_template()`.
+        #
+        #     Then we can simply apply `template_data` to the context in the same layer
+        #     where we apply `context_processor_data` and `component_vars`.
+        with prepare_component_template(component, template_data) as template:
+            # Set `Template._djc_is_component_nested` based on whether we're currently INSIDE
+            # the `{% extends %}` tag.
+            # Part of fix for https://github.com/django-components/django-components/issues/508
+            # See django_monkeypatch.py
+            if template is not None:
+                template._djc_is_component_nested = bool(  # type: ignore[union-attr]
```

Comment:
This `template._djc_is_component_nested` branch is not new, just moved from [`_prepare_template`](https://github.com/django-components/django-components/pull/1222/files#diff-35c84ebceae50f7149f03bbcaf2b4f1289bc2984bf7859ca521b5dbe7960c396L3409)